src/components/strategies/utils.ts

```diff
@@ -35,21 +38,88 @@ export const checkIfOrdersOverlap = (
   orderB: OrderCreate
 ): boolean => {
   if (
-    (+orderB.price <= +orderA.max && +orderB.price !== 0) ||
-    (+orderB.price <= +orderA.price && +orderB.price !== 0) ||
-    (+orderB.min <= +orderA.price && +orderB.min !== 0) ||
-    (+orderB.min <= +orderA.max && +orderB.min !== 0)
+    +orderB.min < +orderA.max &&
+    orderB.max > orderA.max &&
+    +orderB.min > +orderA.min &&
+    +orderB.min !== 0 &&
+    +orderA.min !== 0
   ) {
     return true;
   }
   return false;
 };
 
-export const checkIfOrdersOverlapNew = (
+export const checkIfOrdersReversed = (
+  orderA: OrderCreate, // buy
+  orderB: OrderCreate // sell
+): boolean => {
+  if (
```

Comment:
Would be worth splitting it into multiple conditions. This is difficult to understand